---
title: "Prep. phyloseq database"
author: "Jakub Kreisinger"
date: "May 6, 2021"
output: html_document
---

This script process dada2 outputs and produce phyloseq database

# Packages and custom functions
```{r setup, include=FALSE}
library(dada2)
library(phyloseq)
library(ggplot2)
library(vegan)
library(plyr)
library(dada2)
library(phyloseq)
library(Biostrings)
library(ShortRead)

# this function indentifies PCR duplicates and removes all ASVs that were not present in both of them
dupl.concensus<-function(PHYLOS,NAMES){
  
  # exclude nonduplicated samples
  IDS<-as.character(data.frame(sample_data(PHYLOS))[,NAMES])
  IDS.dupl<-IDS[duplicated(IDS)]
  
  PHYLOSEQ<-prune_samples(IDS%in%IDS.dupl, PHYLOS)
  if(length(IDS.dupl)*2<length(IDS)) {NONUPLICATED<-prune_samples(!IDS%in%IDS.dupl, PHYLOS)
                                       print(paste("Following names are nonduplicated",sample_names(NONUPLICATED)))}
  
  CATS<-as.character(data.frame(sample_data(PHYLOSEQ))[,NAMES])
  CATS2<-levels(factor(CATS))
  OTU_TAB<-otu_table(PHYLOSEQ)
  rownames(OTU_TAB)<-CATS
  
  # i<-5
  for (i in 1:length(CATS2))
  {
    # print(CATS2[i])
    FILTER.act<-colSums(OTU_TAB[rownames(OTU_TAB)==CATS2[i],]>0)>1
    OTU_TAB[rownames(OTU_TAB)==CATS2[i],]
    OTU_TAB[rownames(OTU_TAB)==CATS2[i],]<-t(apply(OTU_TAB[rownames(OTU_TAB)==CATS2[i],],1,function(x) x*FILTER.act))
  }
  
  rownames(OTU_TAB)<-sample_names(PHYLOSEQ)
  otu_table(PHYLOSEQ)<-OTU_TAB
  PHYLOSEQ.clean<-prune_taxa(taxa_sums(PHYLOSEQ)>0,PHYLOSEQ)
  
  PHYLOSEQ.clean
}


#the same as above, but works also for PCR triplicates
dupl.concensus_v2<-function(PHYLOS,NAMES,TRESHOLD=2){
  
  # exclude nonduplicated samples
  IDS<-as.character(data.frame(sample_data(PHYLOS))[,NAMES])
  IDS.dupl<-IDS[duplicated(IDS)]
  
  PHYLOSEQ<-prune_samples(IDS%in%IDS.dupl, PHYLOS)
  if(length(IDS.dupl)*2<length(IDS)) {NONUPLICATED<-prune_samples(!IDS%in%IDS.dupl, PHYLOS)
  print(paste("Following names are nonduplicated",sample_names(NONUPLICATED)))}
  
  CATS<-as.character(data.frame(sample_data(PHYLOSEQ))[,NAMES])
  CATS2<-levels(factor(CATS))
  OTU_TAB<-otu_table(PHYLOSEQ)
  rownames(OTU_TAB)<-CATS
  
  # i<-5
  for (i in 1:length(CATS2))
  {
    # print(CATS2[i])
    FILTER.act<-colSums(OTU_TAB[rownames(OTU_TAB)==CATS2[i],]>0)>=TRESHOLD
    OTU_TAB[rownames(OTU_TAB)==CATS2[i],]
    OTU_TAB[rownames(OTU_TAB)==CATS2[i],]<-t(apply(OTU_TAB[rownames(OTU_TAB)==CATS2[i],],1,function(x) x*FILTER.act))
  }
  
  rownames(OTU_TAB)<-sample_names(PHYLOSEQ)
  otu_table(PHYLOSEQ)<-OTU_TAB
  PHYLOSEQ.clean<-PHYLOSEQ
  # PHYLOSEQ.clean<-prune_taxa(taxa_sums(PHYLOSEQ)>0,PHYLOSEQ)
  
  PHYLOSEQ.clean
}

# this function merge technical replicates in phyloseq object
merge.duplicates<-function(PHYLOSEQ,NAMES){
  CATS<-as.character(data.frame(sample_data(PHYLOSEQ))[,NAMES])
  sample_data(PHYLOSEQ)$duplic.id<-CATS
  SAMDAT<-sample_data(PHYLOSEQ)
  SAMDAT.sub<-subset(SAMDAT,duplicated(CATS)==F)
  FASTA<-refseq(PHYLOSEQ)
  rownames(SAMDAT.sub)<-SAMDAT.sub$duplic.id
  PHYLOSEQ.merge<-merge_samples(PHYLOSEQ,"duplic.id")
  sample_data(PHYLOSEQ.merge)<-SAMDAT.sub
  PHYLOSEQ.merge<-merge_phyloseq(PHYLOSEQ.merge,FASTA)
  PHYLOSEQ.merge
}

```

##############################
# Analyses for the second run# 
##############################
(including tropical samples collected during dry season + trans-Saharan migrants captured in Africa)

## Dada2 abundance matrix to phyloseq-class otu_table
```{r }

PATH<-"/media/kreising/DATA/data/Ptaci_RUN_24.10.2016/Run41_CBGP_JFM/03.DADA.res_NEW/"
list.files()

load(paste(PATH,"otutab_16s.R",sep=""))
dim(seqtab)
seqtab<-otu_table(seqtab,taxa_are_rows = F)

```

## Adjust read orientation

each sample was sequenced in 'normal' and in reverse complement orientation. These lines takes [1] reverse complement samples (sufix "-rc"), [2] reverse complement corresponding ASVs, [3] and merge them with sample sequenced in normal orientation.
```{r }

sample_names(seqtab)
FILTER.rc<-regexpr("-rc-",sample_names(seqtab))>0
seqtab.normal<-prune_samples(FILTER.rc==F,seqtab)
seqtab.normal<-prune_taxa(taxa_sums(seqtab.normal)>0,seqtab.normal)

seqtab.revcomp<-prune_samples(FILTER.rc==T,seqtab)
seqtab.revcomp<-prune_taxa(taxa_sums(seqtab.revcomp)>0,seqtab.revcomp)

dim(seqtab.normal)
dim(seqtab.revcomp)

intersect(taxa_names(seqtab.normal),taxa_names(seqtab.revcomp))

REVCOMP<-reverseComplement(DNAStringSet(taxa_names(seqtab.revcomp)))
taxa_names(seqtab.revcomp)<-as.character(REVCOMP)

length(intersect(taxa_names(seqtab.normal),taxa_names(seqtab.revcomp)))

head(sample_names(seqtab.normal))
head(sample_names(seqtab.revcomp))

sample_names(seqtab.revcomp)<-gsub("-rc","",sample_names(seqtab.revcomp))

seqtab_merged<-merge_phyloseq(seqtab.revcomp,seqtab.normal)
dim(seqtab_merged)
dim(seqtab.revcomp)
dim(seqtab.normal)
```

## Add sample metadata

```{r }
MERGED<-read.delim("/media/kreising/DATA/data/Ptaci_RUN_24.10.2016/Run41_CBGP_JFM/03.DADA.res_NEW/Merged_SD_2run.txt")

NAMES_META<-paste(MERGED$Primer_F,MERGED$Primer_R,"-Lib_",MERGED$Pool,sep="")

#
sum(!NAMES_META%in%sample_names(seqtab_merged))
MERGED[!NAMES_META%in%sample_names(seqtab_merged),]

MERGED<-sample_data(MERGED)
sample_names(MERGED)<-NAMES_META

dim(MERGED)
dim(seqtab_merged)

PHYLOSEQ<-merge_phyloseq(MERGED,seqtab_merged)

```

## Filtering

```{r }
# remove samples that were not sequenced
FILTER<-sample_data(PHYLOSEQ)$sekvenator_1_ano_ne==1
PHYLOSEQ.filt<-prune_samples(FILTER,PHYLOSEQ)
head(sample_data(PHYLOSEQ.filt))

#remove all non-faecal samples
FILTER<-sample_data(PHYLOSEQ.filt)$vzorek=="trus"
FILTER[is.na(FILTER)]<-TRUE
PHYLOSEQ.filt<-prune_samples(FILTER,PHYLOSEQ.filt)

#remove juveniles
FILTER<-sample_data(PHYLOSEQ.filt)$Age.juv.ad.n.%in%c("ad","ad?")
FILTER[is.na(FILTER)]<-TRUE
PHYLOSEQ.filt<-prune_samples(FILTER,PHYLOSEQ.filt)
levels(as.factor(sample_data(PHYLOSEQ.filt)$druh))

#remove samples from P.Prochazka dataset
FILTER<-sample_data(PHYLOSEQ.filt)$state!="CZ"
PHYLOSEQ.filt<-prune_samples(FILTER,PHYLOSEQ.filt)

#remove ASVs with 0 reads
PHYLOSEQ.filt<-prune_taxa(taxa_sums(PHYLOSEQ.filt)>0,PHYLOSEQ.filt)

#Convert ASVs sequence to DNAStringSet object and add it to the phyloseq
REF<-DNAStringSet(taxa_names(PHYLOSEQ.filt))
names(REF)<-DNAStringSet(taxa_names(PHYLOSEQ.filt))
PHYLOSEQ.filt<-merge_phyloseq(REF,PHYLOSEQ.filt)

# Add variables to the sample_data of the phyloseq object
sample_data(PHYLOSEQ.filt)$RUN<-"Ptaci_RUN_24.10.2016"
sample_data(PHYLOSEQ.filt)$ZONE<-"TROPICAL"
```

## Identify PCR dulicates
```{r }
KEEP<-sample_data(PHYLOSEQ.filt)$ID_D[duplicated(sample_data(PHYLOSEQ.filt)$ID_D)]
PHYLOSEQ.DUPL<-prune_samples(sample_data(PHYLOSEQ.filt)$ID_D%in%KEEP,PHYLOSEQ.filt)
PHYLOSEQ.NONDUPL<-prune_samples(!sample_data(PHYLOSEQ.filt)$ID_D%in%KEEP,PHYLOSEQ.filt)
sort(sample_data(PHYLOSEQ.NONDUPL)$ID_D)

DUPL<-duplicated(sample_data(PHYLOSEQ.DUPL)$ID_D)

dupl.1<-prune_samples(DUPL==T,PHYLOSEQ.DUPL)
dupl.2<-prune_samples(DUPL==F,PHYLOSEQ.DUPL)

```

## Check consistency in Shannon alpha diversity
```{r }
DIV1<-estimate_richness(dupl.1, measures = "Shannon")
DIV2<-estimate_richness(dupl.2, measures = "Shannon")

DIV1<-data.frame(DIV1,sample_data(dupl.1))
DIV2<-data.frame(DIV2,sample_data(dupl.2))
DIV1$seqsums1<-sample_sums(dupl.1)
DIV2$seqsums2<-sample_sums(dupl.2)

DIV2<-DIV2[match(DIV1$ID_D, DIV2$ID_D),]

DF<-data.frame(SH1=DIV1$Shannon,
               SH2=DIV2$Shannon,
               NAME=DIV1$ID_D,
               seqsums1=DIV1$seqsums1,
               seqsums2=DIV2$seqsums2,
               sample=DIV1$ID_D)


DF$OUT<-abs(DF$SH1-DF$SH2)<1.2
ggplot(data = DF,aes(x=SH2,y=SH1,color=OUT))+geom_point()

rev(sort(abs(DF$SH1-DF$SH2)))

DF.alpha<-DF
```

## Check consistency in composition
```{r}
library(ape)
BC1<-as.matrix(vegdist(otu_table(transform_sample_counts(dupl.1,function(x) x/sum(x)))))
BC2<-as.matrix(vegdist(otu_table(transform_sample_counts(dupl.2,function(x) x/sum(x)))))
dupl.1.DF<-data.frame(sample_data(dupl.1))
dupl.2.DF<-data.frame(sample_data(dupl.2))
dupl.1.DF$SSUMS<-sample_sums(dupl.1)
dupl.2.DF$SSUMS<-sample_sums(dupl.2)

rownames(dupl.1.DF)<-dupl.1.DF$ID_D
rownames(dupl.2.DF)<-dupl.2.DF$ID_D
dupl.2.DF<-dupl.2.DF[match(rownames(dupl.1.DF),rownames(dupl.2.DF) ),]

rownames(BC1)<-colnames(BC1)<-sample_data(dupl.1)$ID_D
rownames(BC2)<-colnames(BC2)<-sample_data(dupl.2)$ID_D

BC2<-BC2[rownames(BC1),rownames(BC1)]

BC1.pcoa<-pcoa(as.dist(BC1))$vectors
BC2.pcoa<-pcoa(as.dist(BC2))$vectors

PROTEST<-protest(BC1.pcoa,BC2.pcoa)

DF.prot<-data.frame(PROTEST$Y[,1:2],PROTEST$X[,1:2],sample_data(dupl.1),
                    resid=resid(PROTEST))

names(DF.prot)[1:4]<-c("X1","X2","Y1","Y2")

ggplot(DF.prot,aes(x=X1,y=X2))+geom_point()+geom_segment(aes(x=X1,y=X2,xend = Y1, yend = Y2),arrow = arrow(length = unit(0.1,"cm")))
```

## Merge PCR duplicates
```{r}
CONSIST<-dupl.concensus(PHYLOSEQ.DUPL,"ID_D")
CONSIST.merged<-merge.duplicates(PHYLOSEQ=CONSIST,NAMES="ID_D")
sum(otu_table(CONSIST))/sum(otu_table(PHYLOSEQ.DUPL))

(sort(sample_sums(CONSIST.merged)))

sample_data(CONSIST.merged)$DUPLICATED<-"YES"
```

## Add samples, for which the PCR duplicates were not available
```{r}
IN_DUPLICATED<-taxa_names(PHYLOSEQ.NONDUPL)%in%taxa_names(CONSIST.merged)
IN_DUPLICATED<-IN_DUPLICATED*1
FOUR_PRECENT_TAXA<-transform_sample_counts(PHYLOSEQ.NONDUPL,function(x) ifelse((x/sum(x))>0.04,1,0))
unname(otu_table(FOUR_PRECENT_TAXA)[1:5,1:5])

FOUR_PRECENT_TAXA_otutab<-otu_table(FOUR_PRECENT_TAXA)
class(FOUR_PRECENT_TAXA_otutab)<-"matrix"

for(i in 1:dim(FOUR_PRECENT_TAXA_otutab)[2]){
  FOUR_PRECENT_TAXA_otutab[,i]<-(FOUR_PRECENT_TAXA_otutab[,i]+IN_DUPLICATED[i])>0
}

OTU_TAB<-otu_table(PHYLOSEQ.NONDUPL)
OTU_TAB.filtered<-OTU_TAB*FOUR_PRECENT_TAXA_otutab
sum(OTU_TAB)

OTU_TAB.filtered<-otu_table(OTU_TAB.filtered,taxa_are_rows = F)
PHYLOSEQ.NONDUPL2<-PHYLOSEQ.NONDUPL
otu_table(PHYLOSEQ.NONDUPL2)<-OTU_TAB.filtered
PHYLOSEQ.NONDUPL2<-prune_taxa(taxa_sums(PHYLOSEQ.NONDUPL2)>0,PHYLOSEQ.NONDUPL)
sum(otu_table(PHYLOSEQ.NONDUPL))
sum(otu_table(PHYLOSEQ.NONDUPL2))

sample_names(PHYLOSEQ.NONDUPL2)<-sample_data(PHYLOSEQ.NONDUPL)$ID_D
sample_data(PHYLOSEQ.NONDUPL2)$DUPLICATED<-"NO"
sample_data(CONSIST.merged)$DUPLICATED<-"YES"

PHYLOSEQ_dupl_nondulp<-merge_phyloseq(CONSIST.merged,PHYLOSEQ.NONDUPL2)
PHYLOSEQ_tropical2<-PHYLOSEQ_dupl_nondulp

save(PHYLOSEQ_tropical2,file = "/media/kreising/DATA/data/Ptaci_RUN_24.10.2016/Run41_CBGP_JFM/03.DADA.res_NEW/PHYLOSEQ_tropical2.R")
```

##################################################################################
# Processing first run (temperate samples and tropical samples from rainy season)#
##################################################################################
```{r }
PATH<-"/media/kreising/DATA/data/BIRD_MICROBIOME/PASSERINES_COMPARATIVE/FIRST_RUN/DATA_RESULTS_NEW/"
list.files()

load(paste(PATH,"otutab_16s.R",sep=""))
dim(seqtab)
seqtab<-otu_table(seqtab,taxa_are_rows = F)
```

## Adjust read orientation
```{r }
sample_names(seqtab)
FILTER.rc<-regexpr("-rc-",sample_names(seqtab))>0
seqtab.normal<-prune_samples(FILTER.rc==F,seqtab)
seqtab.normal<-prune_taxa(taxa_sums(seqtab.normal)>0,seqtab.normal)

seqtab.revcomp<-prune_samples(FILTER.rc==T,seqtab)
seqtab.revcomp<-prune_taxa(taxa_sums(seqtab.revcomp)>0,seqtab.revcomp)

dim(seqtab.normal)
dim(seqtab.revcomp)

intersect(taxa_names(seqtab.normal),taxa_names(seqtab.revcomp))

REVCOMP<-reverseComplement(DNAStringSet(taxa_names(seqtab.revcomp)))
taxa_names(seqtab.revcomp)<-as.character(REVCOMP)

length(intersect(taxa_names(seqtab.normal),taxa_names(seqtab.revcomp)))

head(sample_names(seqtab.normal))
head(sample_names(seqtab.revcomp))

sample_names(seqtab.revcomp)<-gsub("-rc","",sample_names(seqtab.revcomp))

seqtab_merged<-merge_phyloseq(seqtab.revcomp,seqtab.normal)
dim(seqtab_merged)
dim(seqtab.revcomp)
dim(seqtab.normal)
```

## Add sample metadata
```{r }
MERGED<-read.delim("/media/kreising/DATA/data/Ptaci_RUN_24.10.2016/Run41_CBGP_JFM/03.DADA.res_NEW/Merged_SD_1run.txt")

NAMES_META<-paste(MERGED$primer_F,MERGED$primery_R,"-LIB",
                  gsub("lib","",MERGED$Lib),sep="")

sum(!NAMES_META%in%sample_names(seqtab_merged))
MERGED[!NAMES_META%in%sample_names(seqtab_merged),]

MERGED<-sample_data(MERGED)
sample_names(MERGED)<-NAMES_META

dim(MERGED)
dim(seqtab_merged)

PHYLOSEQ<-merge_phyloseq(MERGED,seqtab_merged)

DF.summary<-data.frame(SSUMS=sample_sums(PHYLOSEQ),
               sample_data(PHYLOSEQ))

DF.summary <- DF.summary[order(DF.summary$SSUMS),] 

write.table(DF,file = "/media/kreising/DATA/data/BIRD_MICROBIOME/PASSERINES_COMPARATIVE/FIRST_RUN/DATA_RESULTS_NEW/SUMMARY.txt",
            sep="\t",row.names = F,quote = F)

PHYLOSEQ.dupl_samples<-PHYLOSEQ
save(PHYLOSEQ.dupl_samples,file="/media/kreising/DATA/data/BIRD_MICROBIOME/PASSERINES_COMPARATIVE/FIRST_RUN/DATA_RESULTS_NEW/PHYLOSEQ.dupl_samples.R")
```

## Filter samples that are not used in the statistical analyses

```{r }
PHYLOSEQ.filt<-PHYLOSEQ

# remove juveniles
FILTER<-sample_data(PHYLOSEQ.filt)$age%in%c("ad","ad?")
FILTER[is.na(FILTER)]<-TRUE
PHYLOSEQ.filt<-prune_samples(FILTER,PHYLOSEQ.filt)
levels(as.factor(sample_data(PHYLOSEQ.filt)$species))

#delete selected species (non-Avian or not free-living)
DELETE_SPECIES<-c("Aplopelia_larvata","Taeniopygia_guttata","Fringilla_montifringilla",
                  "Sasia_africana","Turtur_tympanistria")

# delete nonavian samples
DELETE_SAMPLES<-c("OTAZNIK","Kontrola_A","Kontrola_B","Kontrola_negativni","Kontrola_Negativní","Kontrola_pozitivni",
                  "Kontrola_Pozitivní")

#adjust selected species
PHYLOSEQ.filt<-prune_samples(!sample_data(PHYLOSEQ.filt)$species%in%DELETE_SAMPLES,PHYLOSEQ.filt)
PHYLOSEQ.filt<-prune_samples(!sample_data(PHYLOSEQ.filt)$species%in%DELETE_SPECIES,PHYLOSEQ.filt)
sample_data(PHYLOSEQ.filt)$species<-gsub("Alethe_diademata _OTAZNIK","Alethe_diademata",sample_data(PHYLOSEQ.filt)$species)
sample_data(PHYLOSEQ.filt)$species<-gsub("Alethe_diademata_OTAZNIK","Alethe_diademata",sample_data(PHYLOSEQ.filt)$species)

#remove ASVs with 0 reads
PHYLOSEQ.filt<-prune_taxa(taxa_sums(PHYLOSEQ.filt)>0,PHYLOSEQ.filt)

#Add ASVs reference sequences
REF<-DNAStringSet(taxa_names(PHYLOSEQ.filt))
names(REF)<-DNAStringSet(taxa_names(PHYLOSEQ.filt))
PHYLOSEQ.filt<-merge_phyloseq(REF,PHYLOSEQ.filt)

#Add variables to the phyloseq
sample_data(PHYLOSEQ.filt)$RUN<-"FIRST_RUN"
sample_data(PHYLOSEQ.filt)$ZONE<-ifelse(sample_data(PHYLOSEQ.filt)$state=="CAM","TROPICAL","TEMPERATE")

#remove one sample that was accidentally processed twice
DELETE<-c("BJ05288_Pseudoalcippe_abyssinica_12.09.CE5775CE5775_trus_1","BJ05288_Pseudoalcippe_abyssinica_12.09.CE5775CE5775_trus_2")
PHYLOSEQ.filt<-prune_samples(!sample_data(PHYLOSEQ.filt)$ID%in%DELETE,PHYLOSEQ.filt)

#remove samples with low number of reads
PHYLOSEQ.filt<-prune_samples(sample_sums(PHYLOSEQ.filt)>500,PHYLOSEQ.filt)
```

## Merge technical triplicates 

-a subset of samples was prepred in technical (PCR) triplicates....
-this section identifies triplicated samples and merges them
```{r }
sample_data(PHYLOSEQ.filt)$ID_D<-paste(sample_data(PHYLOSEQ.filt)$species,sample_data(PHYLOSEQ.filt)$ring,sep="_")
SUMMARY<-summary(as.factor(sample_data(PHYLOSEQ.filt)$ID_D),1000)
SUMMARY.3<-names(SUMMARY)[SUMMARY>2]

TRIPLICATES<-prune_samples(sample_data(PHYLOSEQ.filt)$ID_D%in%SUMMARY.3,PHYLOSEQ.filt)

TRIPL_CONC<-dupl.concensus_v2(TRIPLICATES,NAMES="ID_D",TRESHOLD=2)
sum(otu_table(TRIPL_CONC))/sum(otu_table(TRIPLICATES))

MERGED_TRIPL<-merge.duplicates(PHYLOSEQ = TRIPL_CONC,NAMES = "ID_D")
MERGED_TRIPL<-prune_taxa(taxa_sums(MERGED_TRIPL)>0,MERGED_TRIPL)
```

## Identify technical duplicates
```{r }
NON_TRIPL<-prune_samples(!sample_data(PHYLOSEQ.filt)$ID_D%in%sample_data(TRIPLICATES)$ID_D,PHYLOSEQ.filt)
FILTER1<-sample_data(NON_TRIPL)$ID_D%in%VYHOD
FILTER2<-regexpr("LIB11",sample_names(NON_TRIPL))>1
FILTER<-FILTER1+FILTER2!=2
NON_TRIPL<-prune_samples(FILTER,NON_TRIPL)

KEEP<-sample_data(NON_TRIPL)$ID_D[duplicated(sample_data(NON_TRIPL)$ID_D)]
PHYLOSEQ.DUPL<-prune_samples(sample_data(NON_TRIPL)$ID_D%in%KEEP,NON_TRIPL)
PHYLOSEQ.NONDUPL<-prune_samples(!sample_data(NON_TRIPL)$ID_D%in%KEEP,NON_TRIPL)
sort(sample_data(NON_TRIPL)$ID_D)

DUPL<-duplicated(sample_data(PHYLOSEQ.DUPL)$ID_D)

dupl.1<-prune_samples(DUPL==T,PHYLOSEQ.DUPL)
dupl.2<-prune_samples(DUPL==F,PHYLOSEQ.DUPL)
```

## Consistnecy in Shannon diversity
```{r }
DIV1<-estimate_richness(dupl.1, measures = "Shannon")
DIV2<-estimate_richness(dupl.2, measures = "Shannon")

DIV1<-data.frame(DIV1,sample_data(dupl.1))
DIV2<-data.frame(DIV2,sample_data(dupl.2))
DIV1$seqsums1<-sample_sums(dupl.1)
DIV2$seqsums2<-sample_sums(dupl.2)

DIV2<-DIV2[match(DIV1$ID_D, DIV2$ID_D),]

DF<-data.frame(SH1=DIV1$Shannon,
               SH2=DIV2$Shannon,
               NAME=DIV1$ID_D,
               seqsums1=DIV1$seqsums1,
               seqsums2=DIV2$seqsums2,
               sample=DIV1$ID_D)

DF$OUT<-abs(DF$SH1-DF$SH2)<1.2
ggplot(data = DF,aes(x=SH2,y=SH1,color=OUT))+geom_point()

rev(sort(abs(DF$SH1-DF$SH2)))

DF.alpha<-DF
```

## Consistency in microbiota composition between techical duplicates

```{r}
library(ape)
BC1<-as.matrix(vegdist(otu_table(transform_sample_counts(dupl.1,function(x) x/sum(x)))))
BC2<-as.matrix(vegdist(otu_table(transform_sample_counts(dupl.2,function(x) x/sum(x)))))
dupl.1.DF<-data.frame(sample_data(dupl.1))
dupl.2.DF<-data.frame(sample_data(dupl.2))
dupl.1.DF$SSUMS<-sample_sums(dupl.1)
dupl.2.DF$SSUMS<-sample_sums(dupl.2)

rownames(dupl.1.DF)<-dupl.1.DF$ID_D
rownames(dupl.2.DF)<-dupl.2.DF$ID_D
dupl.2.DF<-dupl.2.DF[match(rownames(dupl.1.DF),rownames(dupl.2.DF) ),]

rownames(BC1)<-colnames(BC1)<-sample_data(dupl.1)$ID_D
rownames(BC2)<-colnames(BC2)<-sample_data(dupl.2)$ID_D

BC2<-BC2[rownames(BC1),rownames(BC1)]

BC1.pcoa<-pcoa(as.dist(BC1))$vectors
BC2.pcoa<-pcoa(as.dist(BC2))$vectors

PROTEST<-protest(BC1.pcoa,BC2.pcoa)

DF.prot<-data.frame(PROTEST$Y[,1:2],PROTEST$X[,1:2],sample_data(dupl.1),
                    resid=resid(PROTEST))

names(DF.prot)[1:4]<-c("X1","X2","Y1","Y2")

ggplot(DF.prot,aes(x=X1,y=X2))+geom_point()+geom_segment(aes(x=X1,y=X2,xend = Y1, yend = Y2),arrow = arrow(length = unit(0.1,"cm")))
```

## Merge/filter duplicates
```{r}
CONSIST<-dupl.concensus(PHYLOSEQ.DUPL,"ID_D")
CONSIST.merged<-merge.duplicates(PHYLOSEQ=CONSIST,NAMES="ID_D")
sum(otu_table(CONSIST))/sum(otu_table(PHYLOSEQ.DUPL))

(sort(sample_sums(CONSIST.merged)))

sample_data(CONSIST.merged)$DUPLICATED<-"YES"
```

## Add samples without PCR duplicates

- keep ASVs the were represented by > 4% reads in given sample and/or were confirmed in the set of duplicated PCD samples 
```{r}
IN_DUPLICATED<-taxa_names(PHYLOSEQ.NONDUPL)%in%unique(c(taxa_names(CONSIST.merged),taxa_names(MERGED_TRIPL)))
IN_DUPLICATED<-IN_DUPLICATED*1
FOUR_PRECENT_TAXA<-transform_sample_counts(PHYLOSEQ.NONDUPL,function(x) ifelse((x/sum(x))>0.04,1,0))
unname(otu_table(FOUR_PRECENT_TAXA)[1:5,1:5])

FOUR_PRECENT_TAXA_otutab<-otu_table(FOUR_PRECENT_TAXA)
class(FOUR_PRECENT_TAXA_otutab)<-"matrix"

for(i in 1:dim(FOUR_PRECENT_TAXA_otutab)[2]){
  FOUR_PRECENT_TAXA_otutab[,i]<-(FOUR_PRECENT_TAXA_otutab[,i]+IN_DUPLICATED[i])>0
}

OTU_TAB<-otu_table(PHYLOSEQ.NONDUPL)
OTU_TAB.filtered<-OTU_TAB*FOUR_PRECENT_TAXA_otutab
sum(OTU_TAB)

OTU_TAB.filtered<-otu_table(OTU_TAB.filtered,taxa_are_rows = F)
PHYLOSEQ.NONDUPL2<-PHYLOSEQ.NONDUPL
otu_table(PHYLOSEQ.NONDUPL2)<-OTU_TAB.filtered
PHYLOSEQ.NONDUPL2<-prune_taxa(taxa_sums(PHYLOSEQ.NONDUPL2)>0,PHYLOSEQ.NONDUPL)
sum(otu_table(PHYLOSEQ.NONDUPL))
sum(otu_table(PHYLOSEQ.NONDUPL2))

sample_names(PHYLOSEQ.NONDUPL2)<-sample_data(PHYLOSEQ.NONDUPL)$ID_D
sample_data(PHYLOSEQ.NONDUPL2)$DUPLICATED<-"NO"
sample_data(CONSIST.merged)$DUPLICATED<-"YES"

PHYLOSEQ_dupl_nondulp<-merge_phyloseq(CONSIST.merged,PHYLOSEQ.NONDUPL2,MERGED_TRIPL)
PHYLOSEQ_tropical1<-PHYLOSEQ_dupl_nondulp

save(PHYLOSEQ_tropical1,
     file = "/media/kreising/DATA/data/BIRD_MICROBIOME/PASSERINES_COMPARATIVE/FIRST_RUN/DATA_RESULTS_NEW/PHYLOSEQ_tropical1.R")
```

## Merge phyloseq for the first and the second run 

```{r}
names(sample_data(PHYLOSEQ_tropical1))
names(sample_data(PHYLOSEQ_tropical2))
names(sample_data(PHYLOSEQ_tropical2))[names(sample_data(PHYLOSEQ_tropical2))=="druh"]<-"species"

#Keep subset of variables that are present in both runs
BOTH<-c("species","RUN","ZONE","ID_D","DUPLICATED","duplic.id") 

PHYLOSEQ_tropical1_sub<-PHYLOSEQ_tropical1
sample_data(PHYLOSEQ_tropical1_sub)<-sample_data(PHYLOSEQ_tropical1)[,names(sample_data(PHYLOSEQ_tropical1))%in%BOTH]

PHYLOSEQ_tropical2_sub<-PHYLOSEQ_tropical2
sample_data(PHYLOSEQ_tropical2_sub)<-sample_data(PHYLOSEQ_tropical2)[,names(sample_data(PHYLOSEQ_tropical2))%in%BOTH]

# merge phyloseq
PHYLOSEQ_tropical12<-merge_phyloseq(PHYLOSEQ_tropical1_sub,PHYLOSEQ_tropical2_sub)

# save final phyloseq
save(PHYLOSEQ_tropical12,
     file = "/media/kreising/DATA/data/TROPICAL_TEMPERATE_Birds_2021/data/PHYLOSEQ_tropical12.R")

```








